### PR TITLE
Adding Contributor Guidelines Document

### PR DESCRIPTION
**Pull request by kkeirstead**
Based on https://github.com/dotnet/runtime/blob/d4ac72106c03781c4b458b04f37763ffda044f3e/CONTRIBUTING.md. I'll leave a few comments on the doc itself for areas where I wasn't sure what guidance we should/shouldn't give.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+* A _minimal reproduction_, i.e. the smallest size of code/configuration required to reproduce the wrong behavior.
+* A description of the _expected behavior_, contrasted with the _actual behavior_ observed.
+* Information on the environment: OS/distro, CPU arch, SDK version, etc.
+* Additional information, e.g. is it a regression from previous versions? are there any known workarounds?
+
+### DOs and DON'Ts
+
+Please do:
+
+* **DO** follow our coding style.
```
Comment (kkeirstead) [inline CONTRIBUTING.md:29]: The original doc linked to a style guide to follow - is that something we should add, or is that unnecessary for now?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+* **DO** include tests when adding new features. When fixing bugs, start with
+  adding a test that highlights how the current behavior is broken.
+* **DO** keep the discussions focused. When a new or related topic comes up
+  it's often better to create a new issue than to side track the discussion.
+* **DO** feel free to blog, tweet, or share anywhere else about your contributions!
```
Comment (kkeirstead) [inline CONTRIBUTING.md:34]: We had talked about whether we wanted to have this - I modified it slightly so it sounds less like a requirement.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+5. Add new tests corresponding to your change, if applicable.
+6. Build the repository with your changes.
+    - Make sure that the builds are clean.
+    - Make sure that the tests are all passing, including your new tests.
+7. Create a pull request (PR) against the dotnet/dotnet-monitor repository's **main** branch.
+    - State in the description what issue or improvement your change is addressing.
+    - Check if all the tests are passing.
+8. Wait for feedback or approval of your changes from the team.
+9. When the team has signed off, and all checks are green, your PR will be merged.
+    - The next official build will include your change.
```
[the remark this answers is not in the document]
Comment (jander-msft) [reply CONTRIBUTING.md:68]: Most contributors do not have access to backporting so I wouldn't add that. I think the maintainers would be able to determine whether something should be backported or not and do it themselves. The should also help contributors to scope their changes such that it is backport-able.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+* **DO** include tests when adding new features. When fixing bugs, start with
+  adding a test that highlights how the current behavior is broken.
+* **DO** keep the discussions focused. When a new or related topic comes up
+  it's often better to create a new issue than to side track the discussion.
+* **DO** feel free to blog, tweet, or share anywhere else about your contributions!
+
+Please do not:
+
+* **DON'T** make PRs for style changes.
+* **DON'T** surprise us with big pull requests. Instead, file an issue and start
```
[the remark this answers is not in the document]
Comment (wiktork) [reply CONTRIBUTING.md:39]: Maybe we can be more explicit on when to use each one? For example, I would expect bugs to be created as issues and new features or big changes would makes sense as discussions.

**Review by wiktork — APPROVED** [review]

**Review by wiktork — APPROVED** [review]

**Pull request merged**